src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,61 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS = 10;
+    private final Pair[] pairs;
+    private int size;
+
+    public StorageImpl() {
+        this.pairs = new Pair[MAX_ITEMS];
+    }
+
     @Override
     public void put(K key, V value) {
+        Pair<K, V> existingStorageItem = getStorageItemByKey(key);
+        if (existingStorageItem != null) {
+            existingStorageItem.value = value;
+            return;
+        }
+        Pair<K, V> storageItem = new Pair<>(key, value);
+        if (size == pairs.length) {
+            throw new RuntimeException("Storage is FULL, you can't add new pair");
+        }
```

Comment:
make check on size before create new Pair
```suggestion
        if (size == pairs.length) {
            throw new RuntimeException("Storage is FULL, you can't add new pair");
        }
        Pair<K, V> storageItem = new Pair<>(key, value);
```